helix-core/src/main/java/org/apache/helix/controller/GenericHelixController.java

```diff
@@ -1360,8 +1390,20 @@ private boolean updateControllerState(NotificationContext changeContext, PauseSi
         String uid = UUID.randomUUID().toString().substring(0, 8);
         ClusterEvent event = new ClusterEvent(_clusterName, ClusterEventType.Resume,
             String.format("%s_%s", uid, Pipeline.Type.DEFAULT.name()));
+
+        HelixManager manager = changeContext.getManager();
```

Comment:
Seems we have multiple places to check this event session ID matches. In pushEventToQueue, in HandleEvent(), in ForceRebalance().  Do we really need to validate this everywhere?  Can we combine them together in one place?

Reply:
As mentioned above, they are different: ForceRebalance() and pushEventToQueue() are creating new events. We want to check the leadership and add the session to the event.

For later checks such as in HandleEvent(), they are just using the event session id to check current controller session. If controller session changes, we stop the event/pipeline.

So, since we have 3 places that create new events, I don't think we could combine them together in one place(I've thought about it and tried).

Reply:
To make code clean, I decided to put it the check in one place in handleEvent(). It is fine to push stale events into queue.